remote_version.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+
+	"github.com/hansmi/paperhooks/pkg/client"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type remoteVersionClient interface {
+	GetRemoteVersion(ctx context.Context) (*client.RemoteVersion, *client.Response, error)
+}
+
+type remoteVersionCollector struct {
+	cl remoteVersionClient
+
+	updateAvailableDesc *prometheus.Desc
+}
+
+func newRemoteVersionCollector(cl remoteVersionClient) *remoteVersionCollector {
+	return &remoteVersionCollector{
+		cl: cl,
+
+		updateAvailableDesc: prometheus.NewDesc("paperless_remote_version_update_available", "Whether an update is available.", []string{"version"}, nil),
+	}
+}
+
+func (c *remoteVersionCollector) describe(ch chan<- *prometheus.Desc) {
+	ch <- c.updateAvailableDesc
+}
+
+func (c *remoteVersionCollector) collect(ctx context.Context, ch chan<- prometheus.Metric) error {
+	remoteVersion, _, err := c.cl.GetRemoteVersion(ctx)
+	if err != nil {
+		return err
```

Comment:
We can leave it for this PR, but I'd prefer a remote version failure to not cause a whole scrape to fail. There should be a separate `_success` metric, similar to `probe_success` from [blackbox_exporter](https://github.com/prometheus/blackbox_exporter). A failing probe there does not mean the scrape itself failed.

Reply:
I like the proposal. I created https://github.com/hansmi/prometheus-paperless-exporter/issues/65 to don't forget it.